src/Misc/layoutroot/run.cmd

```diff
@@ -31,8 +31,25 @@ if /i "%~1" equ "localRun" (
     rem ********************************************************************************
     "%~dp0bin\Agent.Listener.exe" %*
 ) else (
-  rem ********************************************************************************
-  rem Run.
-  rem ********************************************************************************
-  "%~dp0bin\Agent.Listener.exe" run %*
+  if /i "%~1" equ "--once" (
+    GOTO RUNONCE
+:SLEEP
+    TIMEOUT /t 5 /nobreak > NUL
+    GOTO RUNONCE
+:RUNONCE
+    rem ********************************************************************************
+    rem Run once.
+    rem ********************************************************************************
+    "%~dp0bin\Agent.Listener.exe" run %*
+    IF %ERRORLEVEL% EQU 3 (
```

Comment:
4

Reply:
consider eliminate loop (failsafe for server health)

its unfortunate the timeout is sealed (cmd file never updates). it would be neat if the agent update laid down a new cmd file too.... except I think that causes a problem (batch file read one line at a time).

another option would be to keep this file simple.